cpp/src/arrow/compute/exec/aggregate_node.cc

```diff
@@ -169,17 +206,63 @@ void AggregatesToString(std::stringstream* ss, const Schema& input_schema,
   *ss << ']';
 }
 
+template <typename BatchHandler>
+Status HandleSegments(std::unique_ptr<GroupingSegmenter>& segmenter,
+                      const ExecBatch& batch, const std::vector<int>& ids,
+                      const BatchHandler& handle_batch) {
+  int64_t offset = 0;
+  ARROW_ASSIGN_OR_RAISE(auto segment_batch, batch.SelectValues(ids));
+  while (true) {
+    ARROW_ASSIGN_OR_RAISE(auto segment, segmenter->GetNextSegment(segment_batch, offset));
+    if (segment.offset >= segment_batch.length) break;
```

Comment:
How can this happen?

Reply:
This is how `GetNextSegment` indicates that it didn't find a next segment. I didn't think adding a flag to indicate this was worth the while.

Reply:
That's fine and makes sense.  Can you add a comment?

Reply:
Done.